### PR TITLE
Update lessons to use DaVinci v41r2

### LINKED ISSUE #174: Move to v40r2
I'm looking at changing the version to v40r2. I'm running the examples with the new version, and plan to provide a PR once I complete them. Please let me know if there's a reason not to do this. So far it is working fairly well.

### PR DESCRIPTION
**Pull request by chrisburr**
This PR updates the lessons to use DaVinci v41r2 and tested all the relevant steps, except those in [10-davinci-grid.md](https://github.com/lhcb/first-analysis-steps/compare/master...chrisburr:update-davinci-v41r2#diff-ce81614c18dddc563467dbcc36011cc7). I've also updated 15-lhcb-dev.md to use `git lb-.*` instead of `getpack`.

Resolves #174.

### CODE REVIEW

File: 15-lhcb-dev.md [hunk 1]
```diff
@@ -27,20 +27,20 @@ Successfully created the local project DaVinciDev in .
...
   > cd ./DaVinciDev
-  > getpack MyPackage vXrY
+  > git lb-use DaVinci
+  > git lb-checkout DaVinci/vXrY MyPackage
 
 then
 
   > make
   > make test
-  > make QMTestSummary
```
Comment (alexpearce) [inline 15-lhcb-dev.md]: Is this not needed anymore? Does `make test` create the summary page?

File: 15-lhcb-dev.md [hunk 2]
```diff
@@ -55,37 +55,33 @@ Your new development environment won't be very useful without any software to mo
...
 ```
 
-Under the hood, `getpack` will `svn checkout` (≈ `git clone`) the corresponding SVN repository.
-The first argument to `getpack` is the name of the package you want to checkout, while the second argument allows you to choose a specific branch.
-`head` is usually the one one that contains the newest development changes and the one you should commit new changes to.
+Under the hood, `git lb-use` will add the [`Stripping`](https://gitlab.cern.ch/lhcb/Stripping) repository as a remote in git.
+`git lb-checkout` will then perform a partial checkout of the master branch of the Stripping repository, only adding the files under [`Phys/StrippingSelections`](https://gitlab.cern.ch/lhcb/Stripping/tree/master/Phys/StrippingSelections).
 
-You can now modify the `StrippingSelections` package and run `make` to build it with your changes.
+You can now modify the `StrippingSelections` package and run `make purge && make` to build it with your changes.
```
Comment (alexpearce) [inline 15-lhcb-dev.md]: Is the `make purge` always necessary?

**Comment by alexpearce** [comment]
@chrisburr to the rescue! 🚓

File: 15-lhcb-dev.md (hunk 2 above)
Comment (chrisburr) [reply 15-lhcb-dev.md]: I took this from the [twiki page](https://twiki.cern.ch/twiki/bin/view/LHCb/TestStrippingLine?rev=9#Getpack_the_head_version_of_Phys), I don't know if it's ever actually needed.

File: 15-lhcb-dev.md (hunk 1 above)
Comment (chrisburr) [reply 15-lhcb-dev.md]: This is just the output obtained by running `lb-dev --name DaVinciDev DaVinci v41r2` (it seems to be the same in v40r2 as well).

Comment (chrisburr) [reply 15-lhcb-dev.md]: `make && make test && make QMTestSummary` just completed with the error `No rule to make target 'QMTestSummary'.`

Comment (alexpearce) [reply 15-lhcb-dev.md]: OK, I guess the actual `lb-dev` script itself has been updated (along with the default Makefile that's generated).

File: 15-lhcb-dev.md (hunk 2 above)
Comment (alexpearce) [reply 15-lhcb-dev.md]: Hmm 🤔 I have found that things often don't work as I expect if I don't `purge`. I think `purge` is useful if you want some Python files updated, as those aren't refreshed after the first `make` call (unlike C++ files). But I'm not sure.

**Comment by ibab** [comment]
@alexpearce: Is this PR ready from your side?

**Comment by alexpearce** [comment]
Yes, but now it needs rebasing 😊

**Comment by chrisburr** [comment]
No it doesn't 😆

**Comment by alexpearce** [comment]
Quite right 😉 LGTM, thank you!

**Pull request merged**